### ISSUE TITLE
Weird import / bundling issue when using `date-fns`

### ISSUE THREAD

**Issue by vladjerca** [monounity/karma-typescript#398]
If I import a specific method from `date-fns` the module is incorrectly resolved.

eg:

``` import addDays from 'date-fns/addDays'```

File.ts:
```ts
import addDays from 'date-fns/addDays';

export function useDateFns() {
    console.log(addDays(new Date(), 10));
}
```

File.spec.ts:
```ts
import { useDateFns } from './File';

fdescribe('File', () => {
    it('should return void', () => {
        expect(useDateFns()).toBeUndefined();
    });
});
```

The test fails with the following error: 
```
SourceMap position not found for trace: TypeError: addDays_1.default is not a function
```

Here comes the juicy part, running the test with the debugger attached renders this:

![image](https://user-images.githubusercontent.com/6339681/91595279-98094580-e963-11ea-9563-50e93b31617c.png)

This does not hold up when running the application in production, it can only be reproduced via `karma-typescript` 😱

Basically from what I can tell is that the bundler is somehow loading the entire `date-fns` library instead of targeting `date-fns/addDays` specifically.
![image](https://user-images.githubusercontent.com/6339681/91595533-ff26fa00-e963-11ea-91a0-888c26849707.png)

LE:

`karma-typescript`: 5.1.0 (latest as of when this issue was opened)

The only workaround I've found for this ATM is to do a really gross hack in my `test configuration script` 

```
import * as dateFns from 'date-fns';
import addDays from 'date-fns/addDays';

(addDays as any) = dateFns.addDays;
```

Let me know if there's any more information I can provide 👍

**Comment by vladjerca** [comment]
@erikbarke 

If this helps, I've managed to find the source of the issue.

Within the `Resolver.ts:166` at this point the `moduleName` is overwritten from `date-fns/addDays` to `date-fns`.

The cause of this issue is that the regex ends up getting applied on `.../node_modules/date-fns/typings.d.ts` for some reason.

**Comment by erikbarke** [comment]
Are you using [compiler paths](https://www.typescriptlang.org/tsconfig#paths)?

**Comment by vladjerca** [comment]
@erikbarke No, this happens with out of the box configuration.

**Comment by vladjerca** [comment]
@erikbarke Here's a repo that should reproduce the issue: https://github.com/vladjerca/karma-typescript-import-bug
